### ISSUE TITLE
Configurable timeout for auto-formatting on BufWritePre

### ISSUE THREAD

**Issue by boblannon** [neoclide/coc.nvim#2957]
**Is your feature request related to a problem? Please describe.**

Whenever an auto-formatter takes longer than 0.5s to apply a change, the status reports that the formatting was changed, but the buffer remains unchanged.

For example, formatting with `black` times out silently when it's triggered by `BufWritePre` (ie, when `"coc.preferences.formatOnSaveFiletypes": ["python"],` is configured). The status line says "Formatted by black" but the buffer doesn't change. I see this behavior whether I'm using `coc-python` or `coc-pyright`. I guess `black` is just a little slow!

I've found that changing [the hard-coded timeout value in `onBufWritePre` fixes this problem](https://github.com/neoclide/coc.nvim/blob/d09f35455b74033121d25b846e66910e0ac0cca5/src/workspace.ts#L1477-L1480), because it gives `black` more time to finish. I set it to `1000` and everything was fine.

```diff
diff --git a/src/workspace.ts b/src/workspace.ts
index 81b4ef82..7dd78c31 100644
--- a/src/workspace.ts
+++ b/src/workspace.ts
@@ -1477,7 +1477,7 @@ augroup end`
         let timer = setTimeout(() => {
           window.showMessage('Will save handler timeout after 0.5s', 'warning')
           resolve(undefined)
-        }, 500)
+        }, 1000)
         let i = 0
         let called = false
```

**Describe the solution you'd like**

It would be nice to be able to configure the timeout to give things like this more time to complete.

If that's not possible, it would be nice if the failure to change the buffer were noted in the status line, rather than the misleading "Formatted by" message.

**Describe alternatives you've considered**

Forking the source code to change the timeout value manually.

**Additional context**

```
## versions

vim version: NVIM v0.4.4
node version: v12.20.1
coc.nvim version: 0.0.80-5cd38e9462
term: iTerm.app
platform: darwin
```

```
black==20.8b1
```

```
python 3.7.8
```

**Comment by yaegassy** [comment]
It seems that the PR has already been created. https://github.com/neoclide/coc.nvim/pull/2921

**Comment by chemzqm** [comment]
Fixed by #2921
